### PR TITLE
ArcTouch

### PR DESCRIPTION
**Pull request by santanaluiz**

### CODE REVIEW

File: src/com/arctouch/bustouch/activities/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+			@Override
+			public void onClick(View v) {
+				Intent i = new Intent(main, LocationActivity.class);
+				main.startActivity(i);
+			}
+		});
+	}
+	
+	@Override
+	public void receiveListOflinhas(List<Route> routes) {
```
Comment (ghost) [inline src/com/arctouch/bustouch/activities/MainActivity.java:49]: Avoid mixing method names with English and Portuguese and also correct the camel case.

File: src/com/arctouch/bustouch/activities/RoutesListActivity.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	
+	private List<Map<String, String>> parseRoutesToMap(List<Route> routes) {
+		List<Map<String, String>> dataList = new ArrayList<Map<String,String>>();
+		
+		for (Route route : routes) {
+			Map<String, String> map = new HashMap<String, String>();
+			map.put("id", String.valueOf(route.getId()));
+			map.put("nome", route.getDisplayName());
+			// FIXME 
+			map.put("empresa", "");
```
Comment (ghost) [inline src/com/arctouch/bustouch/activities/RoutesListActivity.java:123]: If you want to fix this you should query this endpoint:

`https://dashboard.appglu.com/v1/tables/agencies`

File: src/com/arctouch/bustouch/activities/SearchableActivity.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	 * 
+	 * @return
+	 * 		search text
+	 */
+	String getSearchTextValue();
+	
+	/**
+	 * Receive list of route from Thread
+	 */
+	void receiveListOflinhas(List<Route> linhas);
```
Comment (ghost) [inline src/com/arctouch/bustouch/activities/SearchableActivity.java:20]: Same problem of mixing languages.

File: src/com/arctouch/bustouch/json/model/Route.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	public Agency getAgency() {
+		return agency;
+	}
+
+	public void setAgency(Agency agency) {
+		this.agency = agency;
+	}
+
+	public String getDisplayName() {
+		if (isNotNull(this.shortName) && isNotNull(this.longName))
```
Comment (ghost) [inline src/com/arctouch/bustouch/json/model/Route.java:59]: To meet our code standards you should use brackets here. And sometimes a route won't have its short name so could you rewrite this method to reflect this requirement?

**Comment by arctouch-brunobruggemann** [comment]
I didn't add comments on each line. I just wrote some notes (pros and cons).

App Test
- Buttons doesn’t have a selected state implemented;
- The details doesn’t show Route's name, the list of streets;
- I couldn’t test the map. You used the version 1.0, and I can’t generate an API Key to this version anymore.

Code

Cons:
- You wrote some things in English and others in portuguese;
- Some names for variables/attributes could be better (ex: horActitivy);
- There are some useless methods that are never called (ex: getBtnBuscar). Also, it doesn’t make sense create a getter for a button from an Activity;
- You used some strings directly in code (ex:DialogBuilder.showProgressDialog("Aguarde um momento.\nO sistema está realizando a busca.", "Buscando", this);) when it should be in the String.xml;

Pro:
- The folders structure is very well organized; 
- Comments are being used only for important things;
- You broke long methods in small ones; 
- You created some custom components (ex: Long Press in the map and custom font in a TextView);
- You created a class to handle the use of Dialogs;
- Singleton used;
- Abstract class used for the services implementation;
- Screen’s xml are ok. Only a comment in one xml, that you added more LinearLayouts than necessary. But it isn’t a big deal;
- Methods are small and with a good named (it does what the name says);
- The app didn’t crash with the options “Strict mode enable” and “Don’t keep activities” turned on;

**Comment by ghost** [comment]
About the mixing languages and strings on the code you should localize your resources, setting English as default and have a translated file for Portuguese.

Overall the code looks good, congrats!

**Pull request merged**